pkgs/applications/virtualization/OVMF/default.nix

```diff
@@ -58,15 +62,22 @@ stdenv.mkDerivation (edk2.setup projectDscPath {
     ''}
   '';
 
-  buildPhase = if stdenv.isAarch64 then ''
-      build -n $NIX_BUILD_CORES
-    '' else if seabios == null then ''
-      build -n $NIX_BUILD_CORES ${lib.optionalString secureBoot "-DSECURE_BOOT_ENABLE=TRUE"}
-    '' else ''
-      build -n $NIX_BUILD_CORES -D CSM_ENABLE -D FD_SIZE_2MB ${lib.optionalString secureBoot "-DSECURE_BOOT_ENABLE=TRUE"}
-    '';
+  buildPhase = ''
+    # This is required, even though it is set in target.txt in edk2/default.nix.
+    export EDK2_TOOLCHAIN=GCC49
+
+    # Configures for cross-compiling
+    export ''${EDK2_TOOLCHAIN}_${hostArch}_PREFIX=${stdenv.targetPlatform.config}-
+    export EDK2_HOST_ARCH=${hostArch}
+
+    build \
+      -n $NIX_BUILD_CORES \
+      ${buildFlags} \
```

Comment:
Can you use [`escapeShellArgs`](https://github.com/NixOS/nixpkgs/blob/master/lib/strings.nix#L326) to make this an array instead?